src/Core/Components/Progress/FluentProgress.razor

```diff
@@ -2,7 +2,25 @@
 @inherits FluentComponentBase
 @if (Visible)
 {
-    <fluent-progress @ref=Element
+    if (Stroke != ProgressStrokes.Normal || !string.IsNullOrEmpty(Color) || !string.IsNullOrEmpty(BackgroundColor))
+    {
+        <style>
+            @if (Value == null)
+            {
+                @($" #{Id}::part(progress) {{ {StyleIndeterminate} }}")
+                @($" #{Id}::part(indeterminate-indicator-1) {{ {StyleIndeterminateIndicator} }}")
+                @($" #{Id}::part(indeterminate-indicator-2) {{ {StyleIndeterminateIndicator} }}")
+            }
+            else
+            {
+                @($" #{Id}::part(progress) {{ {StyleProgress} }}")
+                @($" #{Id}::part(determinate) {{ {StyleProgressIndicator} }}")
+            }
+        </style>
```

Comment:
Why are you not using the InlineStyleBuilder for this?

Reply:
That's not compatible with ::part

Reply:
That's...a very good reason then 😀